frontend/lib/pages/authentication_service.dart

```diff
@@ -0,0 +1,101 @@
+import 'package:http/http.dart' as http;
+import 'dart:convert';
+
+class AuthenticationService {
+  static const String baseUrl = 'http://localhost:8080'; // Connecting to url
+
+  Future<bool> register(String username, String password) async {
+    // Check if the username and password match a record in the database
+    bool loginResult = await login(username, password);
+    if (loginResult) {
+      // If login successful, return true to indicate successful login without registration
+      return true;
+    }
+
+    // Fetch all users from the database
+    final response = await http.get(
+      Uri.parse('$baseUrl/user/allUsers'),
+      headers: <String, String>{
+        'Content-Type': 'application/json; charset=UTF-8',
+      },
+    );
+
+    if (response.statusCode == 200) {
+      List<dynamic> users = jsonDecode(response.body);
+
+      // Check if the username already exists
+      bool usernameExists = users.any((user) => user['userName'] == username);
+      if (usernameExists) {
+        // Username exists, now check if the password matches
+        var userData = users.firstWhere((user) => user['userName'] == username);
+        if (userData['password'] == password) {
+          // Username and password match, return true to indicate successful login
+          return true;
+        } else {
+          // Username exists but password doesn't match, return false
+          return false;
+        }
+      } else {
```

Comment:
Instead of having them put in a username that does not exist for registration, you should have a seprate "link" that says "don't have an account? register here"